test/python/quality/test_penalty.py

```diff
@@ -0,0 +1,127 @@
+from pathlib import Path
+
+from src.python.review.inspectors.inspector_type import InspectorType
+from src.python.review.inspectors.issue import BaseIssue, IssueType
+from src.python.review.quality.penalty import categorize, PreviousIssue, Punisher
+
+current_issues = [
+    BaseIssue(
+        file_path=Path("."),
+        line_no=1,
+        column_no=1,
+        description="Possibly misspelt word",
+        origin_class="SC200",
+        inspector_type=InspectorType.FLAKE8,
+        type=IssueType.BEST_PRACTICES,
+    ),
+    BaseIssue(
+        file_path=Path("."),
+        line_no=10,
+        column_no=5,
+        description="Lambda may not be necessary",
+        origin_class="W0108",
+        inspector_type=InspectorType.FLAKE8,
+        type=IssueType.CODE_STYLE,
+    ),
+]
+
+punisher = Punisher([], [])
+
+
+def test_get_penalizing_classes_empty_previous_issues_empty_current_issues():
+    actual = punisher._get_penalizing_classes([], [])
+
+    assert actual == set()
+
+
+def test_get_penalizing_classes_empty_current_issues():
+    actual = punisher._get_penalizing_classes(current_issues, [])
+
+    assert actual == set()
+
+
+def test_get_penalizing_classes_empty_previous_issues():
+    previous_issues = [PreviousIssue("WPS301", 50), PreviousIssue("SC200", 10)]
+
+    actual = punisher._get_penalizing_classes([], previous_issues)
+    assert actual == set()
+
+
+def test_get_penalizing_classes_empty_intersection():
+    previous_issues = [PreviousIssue("WPS301", 50), PreviousIssue("WPS412", 10)]
+
+    actual = punisher._get_penalizing_classes(current_issues, previous_issues)
+    assert actual == set()
+
+
+def test_get_penalizing_classes_non_empty_intersection_1():
```

Comment:
Please, use parameterized tests for several cases with tests with the same logic